python/dgl/sparse/sparse_matrix.py

```diff
@@ -479,6 +479,62 @@ def is_diag(self):
         """Returns whether the sparse matrix is a diagonal matrix."""
         return self.c_sparse_matrix.is_diag()
 
+    def select(self, dim: int, ids: torch.Tensor or slice):
+        """Returns a sub-matrix selected by a list of ids or by range.
+
+        Parameters
+        ----------
+        dim : int
+            The dim to select from matrix, should be 0 or 1.
+        ids : tensor.Tensor or slice
```

Comment:
what's slice? a interger?

Reply:
zhenkun suggest to use slice, a type of python used as (start, end, step). I think it is hard to be recognized whether users want to select two rows (x, y) or the range between them [x, y) if we use Tensor.

Reply:
The range is not a Tensor. It is a [Python Slice](https://www.w3schools.com/python/ref_func_slice.asp) with three arguments: start, stop, and step. So I think it is better to use adopt two interfaces for the slicing, i.e., `index_select(dim, tensor)` and `range_select(dim, slice)`.

Reply:
Ah, I see, I was not aware that slice is a python type, then this API makes sense.